### PR TITLE
Rewards test

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

**Review by goncer — APPROVED** [review]
lgtm

File: test/parallel/xyk-pallet.API.poolLiqidity.activateLiqudity.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*
+ *
+ * @group xyk
+ * @group poolliquidity
+ * @group parallel
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.activateLiqudity.test.ts:5]: remove parallel group plz! 
those tests need fast-runtime env

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
+/*
+ *
+ * @group xyk
+ * @group poolliquidity
+ * @group parallel
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:5]: :pray:

File: test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    token1,
+    new BN(valueBurningTokens)
+  );
+
+  await mangata.claimRewards(
+    testUser1.keyRingPair,
+    liqIdPromPool.toString(),
+    defaultCurrencyValue
+  );
+
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts:142]: plz do the following: 
* Check that the user got some tokes after claiming
* Wait one session
* Run the rpc for calculate rewards
* Validate that is 0.

File: test/parallel/xyk-pallet.API.poolLiqidity.mintLiqudity.test.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
+/*
+ *
+ * @group xyk
+ * @group poolliquidity
+ * @group parallel
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqidity.mintLiqudity.test.ts:5]: :pray:

File: test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
+/*
+ *
+ * @group xyk
+ * @group poolliquidity
+ * @group parallel
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts:5]: :pray:

File: test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    Assets.promotePool(liqIdPromPool.toNumber(), 20),
+    Assets.mintNative(testUser1)
+  );
+
+  testUser1.addAsset(liqIdPromPool);
+  testUser2.addAsset(token2);
+  testUser2.addAsset(liqIdNonPromPool);
+});
+
+test("Check that a user can burn  tokens when they are activated and when burning the free are subtracted first", async () => {
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts:101]: Took me a while understanding the test: plz rename :) 
Check that a user can burn tokens when they are activated, and when burning the free,  those  subtracted first

**Review by goncer — CHANGES_REQUESTED** [review]
Plz review the comments!

File: test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts (hunk 6 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts]: Renamed

File: test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts (hunk 5 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqudity.burnTokens.test.ts]: Deleted

File: test/parallel/xyk-pallet.API.poolLiqidity.mintLiqudity.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqidity.mintLiqudity.test.ts]: Deleted

File: test/parallel/xyk-pallet.API.poolLiqidity.activateLiqudity.test.ts (hunk 1 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.poolLiqidity.activateLiqudity.test.ts]: Deleted

File: utils/utils.ts [hunk 7]
```diff
@@ -281,6 +281,44 @@ export async function waitIfSessionWillChangeInNblocks(numberOfBlocks: number) {
...
+export async function waitNewStakingRound(maxBlocks: number = 0) {
+  let currentSessionNumber: any;
+  let currentBlockNumber: any;
+  let awaitedSessionNumber: any;
+  const api = getApi();
+  const parachainStakingRoundInfo = await api?.query.parachainStaking.round();
+  const sessionLength = parachainStakingRoundInfo.length.toNumber();
+  currentBlockNumber = await getBlockNumber();
+  const initialBlockNumber = currentBlockNumber;
+  currentSessionNumber = await api?.query.session.currentIndex();
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply utils/utils.ts]: Changed type of all const to `number`

**Pull request merged**